core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -211,6 +211,22 @@ oppia.factory('CollectionObjectFactory', [
       return angular.copy(Object.keys(this._explorationIdToNodeIndexMap));
     };
 
+    // Changes the position of node in the nodes list of collection.
+    Collection.prototype.moveCollectionNode = function(expId, newPosition) {
+      if (this._explorationIdToNodeIndexMap.hasOwnProperty(expId)) {
+        var oldPosition = this._explorationIdToNodeIndexMap[expId];
+        var node = angular.copy(this._nodes[oldPosition]);
+
+        if (this._nodes.length > newPosition && newPosition >= 0) {
```

Comment:
We can remove this check if the following functionality is not required:
https://github.com/oppia/oppia/blob/94c3c83825bb190df2376942e0842fc35d88524d/core/templates/dev/head/pages/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js#L303-L306